packages/convai-widget-core/src/contexts/conversation.tsx

```diff
@@ -182,7 +253,14 @@ function useConversationSetup() {
             message = e.message || message;
           }
           error.value = message;
-          transcript.value = [...transcript.value, { type: "error", message }];
+          transcript.value = [
+            ...transcript.value,
+            {
+              type: "error",
+              message,
+              conversationIndex: conversationIndex.peek(),
+            },
+          ];
```

Comment:
Wonder if we can run into some rendering issues when the array is updated. React is not best with those, so might be issue with longer conversations, especially if we want to animate things. 

Updating messages (ex. loading state if something is loading for long time, and eventually maybe fails to send) may be tricky too. Perhaps some message IDs is something to consider?